aten/src/ATen/core/op_registration/kernel_functor.h

```diff
@@ -95,6 +95,22 @@ namespace detail {
     static_assert(guts::false_t<T>::value, "You tried to register a kernel with an unsupported integral argument type. Please use int64_t instead.");
   };
 
+  // legacy_ivalue_to_arg_type is like ivalue_to_arg_type but additionally
+  // allows a few deprecated types like std::vector.
+  template<class T>
+  struct legacy_ivalue_to_arg_type final {
+    static auto call(IValue&& v) -> decltype(ivalue_to_arg_type<T>::call(std::move(v))) {
+      return ivalue_to_arg_type<T>::call(std::move(v));
+    }
+  };
+  template<class T>
+  struct legacy_ivalue_to_arg_type<std::vector<T>> final {
```

Comment:
maybe create similar wrapper for std::vector in non-legacy case and add message to in static assert to do ArrayRef instead

Reply:
that already exists, see line 82.